src/Pim/Bundle/CatalogBundle/ElasticSearch/SearchQueryBuilder.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\ElasticSearch;
+
+/**
+ * TODO: /////////////////////////////////////////////////////////////////////////////////
+ *
+ * For now we call this class "Builder", because that's what it does. It builds iteratively the final ES query
+ * and renders it as a php array when needed. Instances of this class are set in the filters (setQueryBuilder) so that
+ * each filters has the possibility to add its own small condition to the query thanks to this SQB object.
+ *
+ * That is fine, and that's what we did in the previous version. But what we want to do instead is get rid of this
+ * Search Query builder class and have each of the filters capable of modifying the data directly (because they are the
+ * real builder parts in this architecture). So this class only becomes a data holder (which filter are allowed to
+ * modify)and is capable of returning a full featured working query using getQuery(). This SearchQuery could easily
+ * implement a more generic class which interface could support Doctrine ORM Queries, etc.. just something generic
+ * enough.
```

Comment:
Okay, but I would like to see an example. The clauses should be easily accessible by the filter. And it means, you have no way anymore to interact with the query before it gets executed. So it removes some possibilities.